plugins/techdocs/src/home/components/TechDocsCustomHome.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import { useAsync } from 'react-use';
+import { makeStyles } from '@material-ui/core';
+
+import { catalogApiRef, CatalogApi } from '@backstage/plugin-catalog-react';
+import { Entity } from '@backstage/catalog-model';
+import {
+  CodeSnippet,
+  Content,
+  ConfigApi,
+  configApiRef,
+  Header,
+  HeaderTabs,
+  Page,
+  Progress,
+  useApi,
+  WarningPanel,
+  SupportButton,
+  ContentHeader,
+} from '@backstage/core';
+import { DocsTable } from './DocsTable';
+import { DocsCardGrid } from './DocsCardGrid';
+
+const widgets = {
+  DocsTable: DocsTable,
+  DocsCardGrid: DocsCardGrid,
+};
+
+export type WidgetType = 'DocsCardGrid' | 'DocsTable';
+
+export interface WidgetConfig {
+  title: string;
+  description: string;
+  widgetType: WidgetType;
+  widgetMaxHeight?: string;
+  filterPredicate: (entity: Entity) => boolean;
```

Comment:
Thinking about this too...  This makes a lot of sense on a small backstage site with very few documented entities in the catalog.

But on an instance of Backstage with hundreds or thousands of components, we probably don't want to load all of those in the browser and filter them in JS.

May be better to provide something that could be passed as a filter to the catalog client.

Reply:
If by which you mean for backend catalog api to allow querying with filters so that filtering on entities can be done on backend, I agree yes that would be quite important for scalability. More traditional systems would have both filtering and pagination on large collections of items handled at backend or DB.

We can look at that as a separate improvement I think.

Reply:
It doesn't seem to be possible to pass 'only entities with a truthy value for this key' to the catalog client getEntities filter object or other catalog client methods , which is needed for identifying documentation entities ('only entities with any value for the key "backstage.io/techdocs-ref" ').
Please let me know if I'm wrong.

Reply:
Yeah, you're right.  I guess the core team is consciously keeping catalog filtration quite limited, waiting to defer more advanced filter use-cases to search.  Let's stick with this now and evolve.  Thanks for checking!

Reply:
Yep, confirming. Checking for the presence of a field sounds like a very sensible thing to want though, so could probably consider adding support for it